wsmaster/codenvy-hosted-platform-api-impl/src/main/java/com/codenvy/api/workspace/LimitsCheckingWorkspaceManager.java

```diff
@@ -161,6 +186,35 @@ public WorkspaceImpl updateWorkspace(String id, Workspace update) throws Conflic
     }
 
     /**
+     * One of the checks in {@link #checkRamAndPropagateStart(WorkspaceConfig, String, String, String, WorkspaceCallback)}
+     * requests to docker for system ram amount. This request can take much time, so we can not mark the method as synchronized
+     * because this will reduce performance. The solution is to make the system ram check optimistic by limiting
+     * same time start workspace requests. All requests to start workspace performed at the same time can receive not actual
+     * system ram limit status, so limits.workspace.start.throughput property configures how many requests to handle at the same time.
+     * If limits.workspace.start.threads.limit property is set to 0 or less then 0 all requests will be handled at the same time.
```

Comment:
This is a typo actually, as the property name should be `limits.workspace.start_throughput`.
Also i'd operate term s as _parallel method calls_ instead of _requests_